File: julia/p2.jl
function fibs()
  local fibs = [1, 1]

  val = 0
  while val < 4000000
    local next = fibs[end] + fibs[end - 1]
    val = next
    push!(fibs, next)
  end

  fibs
end

function even_fibs(fibs)
  tmp = [mod(i, 2) == 0 ? i : 0 for i=fibs]
end

t1 = time()
for i = 1:10000
  tmp = fibs()
  tmp2 = even_fibs(tmp) 
  sum(tmp2)
end
t2 = time()
println("elapsed time: ", t2 - t1, " seconds")


function sum_of_even_fibs()
  sum = 0
  fibs = [1, 1]

  val = 0
  while val < 4000000
    local next = fibs[end] + fibs[end - 1]
    val = next
    push!(fibs, next)
    if mod(fibs[end], 2) == 0
      sum += fibs[end]
    end
  end

  sum
end

t1 = time()
for i = 1:10000
  sum_of_even_fibs()
end
t2 = time()
println("elapsed time: ", t2 - t1, " seconds")

File: go/p2.go
package main

import(
  "fmt"
  "time"
)

func fibonacci() int {
  fibs := []int{1, 2}
  last := 2
  sum := last

  for last < 4000000 {
    fibsLen := len(fibs)
    last = fibs[fibsLen - 1] + fibs[fibsLen - 2]
    fibs = append(fibs, last)

    if last % 2 == 0 {
      sum += last
    }
  }

  return sum
}

func main() {
  s := time.Now()

  for i := 0; i < 10000; i += 1 {
    fibonacci()
  }

  fmt.Println(time.Now().Sub(s))
}
